### PR TITLE
Fix csi bug. Select node from accessibility requirements if any.

### PR DESCRIPTION
**Pull request by pzheng1025**

### CODE REVIEW

File: csi/src/meta_data.rs [hunk 1]
```diff
@@ -191,60 +192,82 @@ impl MetaData {
...
+    fn get_random_node_from_topology(topology: &[Topology]) -> Option<String> {
+        let mut rng = rand::thread_rng();
+        topology.iter().choose(&mut rng).and_then(|t| {
+            t.get_segments()
+                .values()
+                .choose(&mut rng)
+                .map(std::borrow::ToOwned::to_owned)
+        })
+    }
+    /// Validate if a node exists in topology
```
Comment (pwang7) [inline csi/src/meta_data.rs:205]: add empty line before comments

File: csi/src/meta_data.rs [hunk 2]
```diff
@@ -191,60 +192,82 @@ impl MetaData {
...
+    fn validate_node_in_topology(node_id: &str, topology: &[Topology]) -> bool {
+        topology
+            .iter()
+            .any(|t| t.get_segments().iter().any(|(_, v)| v == node_id))
+    }
+    /// Validate if a node exists in accessibility requirements
```
Comment (pwang7) [inline csi/src/meta_data.rs:211]: add empty line before comments

File: csi/src/meta_data.rs [hunk 3]
```diff
@@ -191,60 +192,82 @@ impl MetaData {
...
+    fn validate_node_in_accessibility_requirements(
+        node_id: &str,
+        access_req: &TopologyRequirement,
+    ) -> bool {
+        let match_requisite = Self::validate_node_in_topology(node_id, access_req.get_requisite());
+        let match_preferred = Self::validate_node_in_topology(node_id, access_req.get_preferred());
+        match_requisite || match_preferred
+    }
     /// Select a node to create volume or snapshot
```
Comment (pwang7) [inline csi/src/meta_data.rs:220]: add empty line before comments

**Review by pwang7 — CHANGES_REQUESTED** [review]

File: csi/src/meta_data.rs (hunk 1 above)
Comment (pzheng1025) [reply csi/src/meta_data.rs:206]: Done

File: csi/src/meta_data.rs (hunk 2 above)
Comment (pzheng1025) [reply csi/src/meta_data.rs:213]: Done

File: csi/src/meta_data.rs (hunk 3 above)
Comment (pzheng1025) [reply csi/src/meta_data.rs:223]: Done

**Review by pwang7 — APPROVED** [review]

**Pull request merged**